packages/astro/src/actions/index.ts

```diff
@@ -0,0 +1,88 @@
+import { existsSync } from 'node:fs';
+import { mkdir, readFile, writeFile } from 'node:fs/promises';
+import type { AstroIntegration } from '../@types/astro.js';
+import { ACTIONS_TYPES_FILE, RESOLVED_VIRTUAL_MODULE_ID, VIRTUAL_MODULE_ID } from './consts.js';
+import type { Plugin as VitePlugin } from 'vite';
+
+export default function astroActions(): AstroIntegration {
+	return {
+		name: VIRTUAL_MODULE_ID,
+		hooks: {
+			async 'astro:config:setup'(params) {
+				const stringifiedActionsPath = JSON.stringify(
+					new URL('actions', params.config.srcDir).pathname
```

Comment:
It's hard to tell how this is being used, but should this be using `fileURLToPath` instead, otherwise I think this will not work on Windows.

Reply:
this is embedding the actions path as an ESM import path. Since this is static and resolved by Vite, I thought `pathname` would be the way to go. Am I correct if that's the intent?

Reply:
Renamed to `stringifiedActionsImport` for clarity